frontend-react/src/components/MessageTracker/MessageReceivers.tsx

```diff
@@ -1,62 +1,479 @@
+    None: "",
+    ASC: "asc",
+    DESC: "desc",
+};
+
+const StatusEnum = {
+    BATCH: "BATCH",
+    PROCESS: "PROCESS",
+    READY: "READY",
+};
+
+const MessageReceiversCol = ({
+    columnHeaderTitle,
+    activeColumn,
+    setActiveColumn,
+    activeColumnSortOrder,
+    setActiveColumnSortOrder,
+    filterIcon,
+    rowSpan,
+}: MessageReceiversColProps) => {
+    const isCurrentlyActiveColumn = columnHeaderTitle === activeColumn;
+    const handleColHeaderClick = () => {
+        if (!isCurrentlyActiveColumn) {
+            // Reset active column and sort order on new column click
+            setActiveColumn(columnHeaderTitle);
+            setActiveColumnSortOrder(FilterOptionsEnum.None);
+        }
+
+        // Explicitly set the proceeding sort order
+        switch (true) {
+            case activeColumnSortOrder === FilterOptionsEnum.None:
+                setActiveColumnSortOrder(FilterOptionsEnum.ASC);
+                break;
+            case activeColumnSortOrder === FilterOptionsEnum.ASC:
+                setActiveColumnSortOrder(FilterOptionsEnum.DESC);
+                break;
+            case activeColumnSortOrder === FilterOptionsEnum.DESC:
+                setActiveColumnSortOrder(FilterOptionsEnum.None);
+                break;
+        }
+    };
+    return (
+        <th
+            className={
+                isCurrentlyActiveColumn &&
+                activeColumnSortOrder !== FilterOptionsEnum.None
+                    ? "active-col-header"
+                    : ""
+            }
+            onClick={handleColHeaderClick}
+            rowSpan={rowSpan}
+        >
+            <div className="column-header-title-container">
+                <p>{columnHeaderTitle}</p>
+                {isCurrentlyActiveColumn ? filterIcon : <Icon.SortArrow />}
+            </div>
+        </th>
+    );
+};
+
+const MessageReceiversRow = ({
+    receiver,
+    activeColumn,
+    activeColumnSortOrder,
+    setModalText,
+    modalRef,
+}: MessageReceiversRowProps) => {
+    // We highlight the entire active column so within the row,
+    // we need to know which column is active
+    const checkForActiveColumn = (colName: string) =>
+        colName === activeColumn &&
+        activeColumnSortOrder !== FilterOptionsEnum.None
+            ? "active-col-td"
+            : "";
+    const uppercaseMain = receiver.Main.toUpperCase();
+    return (
+        <tr>
+            <td className={checkForActiveColumn(ColumnDataEnum.Name)}>
+                {receiver.Name}
+            </td>
+            <td className={checkForActiveColumn(ColumnDataEnum.Service)}>
+                {receiver.Service}
+            </td>
+            <td className={checkForActiveColumn(ColumnDataEnum.Date)}>
+                {receiver.Date}
+            </td>
+            <td
+                className={`message-receiver-break-word ${checkForActiveColumn(
+                    ColumnDataEnum.ReportId
+                )}`}
+                onClick={() => {
```

Comment:
Shouldn't we be truncating the field and then only showing modal if needed?

Reply:
Talked to Jillian and figured a path forwards here!

Reply:
Might be nice to break out this onClick into a function:

```jsx
function handleCellClick(title, body)
    setModalText({ title, body });
    modalRef?.current.toggleModal();
}
```

Then just use that instead of redefining it each time below:

```jsx
<td
    onClick={ () => handleCellClick(ColumnDataEnum.ReportId, receiver.reportId) }
>
```

Reply:
Yup, I like that.